In [8]:
import pandas as pd
import numpy as np

In [13]:
### load data
dateparse = lambda dates: pd.datetime.strptime(dates, '%Y-%m-%d')
users = []
for f_name in glob('Students/*.csv'):
    user = pd.read_csv(f_name, parse_dates=['resp_time'], index_col=['resp_time'],date_parser=dateparse)
    users.append(user)

df = pd.concat(users, ignore_index=False)

df.sort_values(by='resp_time').head()
all_data = df.groupby(level=0).mean()
all_data.head(10)

stress_level  sleep_hour  sleep_rate  exercise   Social2  \
resp_time                                                              
2013-04-01      1.150000    6.666667    2.208333  1.000000  3.000000   
2013-04-02      1.089286    7.125000    2.000000  1.875000  2.250000   
2013-04-03      0.859848    6.750000    2.125000  2.125000  2.000000   
2013-04-04      0.968750    7.750000    1.687500  2.125000  2.125000   
2013-04-05      1.307212    5.687500    1.968750  1.375000  2.437500   
2013-04-06      1.025000    6.750000    1.750000  1.000000  2.750000   
2013-04-07      0.982993    7.857143    1.714286  1.714286  2.071429   
2013-04-08      0.819444    7.041667    1.791667  1.500000  2.262566   
2013-04-09      0.375000    9.000000    1.000000  1.000000  2.833333   
2013-04-10      1.287500    6.687500    1.937500  1.250000  2.135169   

             working  relaxing  social_number   anxious      calm  \
resp_time                                                           
2013-04-01  2.601025  1.596726       2.916667  1.945701  3.912519   
2013-04-02  2.374360  1.624616       3.500000  2.246739  3.389130   
2013-04-03  2.543189  1.675641       3.029167  2.395387  3.586777   
2013-04-04  2.605629  1.553327       3.000000  2.674981  3.460283   
2013-04-05  2.470764  1.429701       3.512500  2.643731  3.591235   
2013-04-06  2.221825  1.432900       3.375000  2.250463  3.728704   
2013-04-07  2.711142  1.432462       2.761905  2.226596  3.582603   
2013-04-08  2.507360  1.807336       2.833333  2.000000  3.291667   
2013-04-09  3.500000  2.000000       2.500000  2.000000  4.000000   
2013-04-10  3.125000  1.375000       3.500000  2.500000  4.000000   

            disorganized  enthusiastic  critical  conventional  
resp_time                                                       
2013-04-01      1.681373      3.162330  1.696833      3.140152  
2013-04-02      2.723913      2.636957  2.217391      2.340217  
2013-04-03      2.222742      2.893096  2.147191      3.271831  
2013-04-04      2.494865      3.018233  2.297710      3.414743  
2013-04-05      2.173436      2.826269  1.878067      3.264445  
2013-04-06      1.591667      2.825463  1.342593      2.742593  
2013-04-07      2.432227      3.098616  2.247566      3.077671  
2013-04-08      1.770833      2.375000  1.979167      2.979167  
2013-04-09      3.000000      3.000000  1.500000      3.750000  
2013-04-10      2.187500      3.125000  1.812500      3.500000

In [18]:
all_data.describe()

stress_level  sleep_hour  sleep_rate   exercise    Social2    working  \
count     48.000000   48.000000   48.000000  48.000000  48.000000  48.000000   
mean       1.305100    8.183135    1.667876   1.564707   2.600003   2.843270   
std        0.740841    2.578248    0.616029   0.935200   0.540446   1.070561   
min        0.000000    1.000000    1.000000   1.000000   1.791667   1.000000   
25%        0.959896    6.737500    1.000000   1.000000   2.145995   2.305456   
50%        1.097024    8.000000    1.732143   1.100000   2.433036   2.875000   
75%        1.833333    9.000000    2.000000   1.703571   2.922738   3.500000   
max        3.000000   17.000000    4.000000   5.000000   3.625000   5.000000   

        relaxing  social_number    anxious       calm  disorganized  \
count  48.000000      48.000000  48.000000  48.000000     48.000000   
mean    1.574883       3.144345   2.444218   3.797538      2.461127   
std     0.611303       0.829372   0.602860   0.731032      0.841081   
min     1.000000       1.500000   1.000000   1.000000      1.000000   
25%     1.000000       2.738095   2.000000   3.500000      2.000000   
50%     1.432681       3.000000   2.333333   4.000000      2.416113   
75%     2.000000       3.503125   2.770833   4.062500      3.000000   
max     4.000000       6.000000   4.000000   5.000000      4.000000   

       enthusiastic   critical  conventional  
count     48.000000  48.000000     48.000000  
mean       3.000286   1.905142      3.259638  
std        0.809504   0.609448      0.607434  
min        1.000000   1.000000      2.000000  
25%        2.500000   1.500000      2.967708  
50%        3.000000   2.000000      3.153409  
75%        3.180081   2.260102      3.750000  
max        5.000000   3.250000      5.000000

# baseline 1: average stress

In [22]:
stress_day_percnt = len(all_data[all_data['stress_level'] > 2.5]) / len(all_data)
stress_day_percnt

0.0625

baseline 1: average stress level <br>
The average student stress level is 1.305. If we set stress level > 2.5 to be stress, and stress level <2.5 to be non-stress, only 6.25% of all surveyed days that students would generally feel stressed.

# baseline 2: Naive bayes Classifier

In [30]:
all_data['stress_level'].head()

resp_time
2013-04-01    1.150000
2013-04-02    1.089286
2013-04-03    0.859848
2013-04-04    0.968750
2013-04-05    1.307212
Name: stress_level, dtype: float64

In [32]:
# set stress = 1
all_data['stress_or_not'] = np.where(all_data['stress_level'] >= 2.5, 1,0)
all_data['stress_or_not']
all_data.head()

stress_level  sleep_hour  sleep_rate  exercise  Social2   working  \
resp_time                                                                       
2013-04-01      1.150000    6.666667    2.208333     1.000   3.0000  2.601025   
2013-04-02      1.089286    7.125000    2.000000     1.875   2.2500  2.374360   
2013-04-03      0.859848    6.750000    2.125000     2.125   2.0000  2.543189   
2013-04-04      0.968750    7.750000    1.687500     2.125   2.1250  2.605629   
2013-04-05      1.307212    5.687500    1.968750     1.375   2.4375  2.470764   

            relaxing  social_number   anxious      calm  disorganized  \
resp_time                                                               
2013-04-01  1.596726       2.916667  1.945701  3.912519      1.681373   
2013-04-02  1.624616       3.500000  2.246739  3.389130      2.723913   
2013-04-03  1.675641       3.029167  2.395387  3.586777      2.222742   
2013-04-04  1.553327       3.000000  2.674981  3.460283      2.494865   
2013-04-05  1.429701       3.512500  2.643731  3.591235      2.173436   

            enthusiastic  critical  conventional  stress_or_not  
resp_time                                                        
2013-04-01      3.162330  1.696833      3.140152              0  
2013-04-02      2.636957  2.217391      2.340217              0  
2013-04-03      2.893096  2.147191      3.271831              0  
2013-04-04      3.018233  2.297710      3.414743              0  
2013-04-05      2.826269  1.878067      3.264445              0

In [41]:
from sklearn.metrics import accuracy_score
from sklearn.model_selection import train_test_split
import sklearn.naive_bayes as sknb

X = all_data.drop(columns = ['stress_level','stress_or_not']).values
y = all_data['stress_or_not']

X_train, X_test, y_train, y_test = train_test_split(X,y)

gnb_model = sknb.GaussianNB()
gnb_model.fit(X_train,y_train)

y_pred = gnb_model.predict(X_test)

accuracy = accuracy_score(y_test, y_pred)
print("Accuracy: {0:.2%}".format(accuracy))

Accuracy: 75.00%
